engines/stark/ui/menu/locationscreen.cpp

```diff
@@ -171,6 +180,10 @@ void StaticLocationWidget::onClick() {
 
 	if (_soundMouseClick) {
 		_soundMouseClick->play();
+		// Ensure the click sound is played completely
+		while (_soundMouseClick->isPlaying()) {
+			g_system->delayMillis(10);
+		};
```

Comment:
You need to add g_system->updateScreen() here too, otherwise you could get artefacts or jerky mouse movements.